src/internal/connector/exchange/api/events.go

```diff
@@ -190,6 +204,66 @@ func (c Events) GetAddedAndRemovedItemIDs(
 	return added, nil, DeltaUpdate{}, errs.ErrorOrNil()
 }
 
+// ---------------------------------------------------------------------------
+// Serialization
+// ---------------------------------------------------------------------------
+
+// Serialize retrieves attachment data identified by the event item, and then
+// serializes it into a byte slice.
+func (c Events) Serialize(
+	ctx context.Context,
+	item serialization.Parsable,
+	user, itemID string,
+) ([]byte, error) {
+	event, ok := item.(models.Eventable)
+	if !ok {
+		return nil, fmt.Errorf("expected Eventable, got %T", item)
+	}
+
+	var (
+		err    error
+		writer = kioser.NewJsonSerializationWriter()
+	)
+
+	defer writer.Close()
+
+	if *event.GetHasAttachments() {
```

Comment:
I know this is just moving code around, but this feels more like it should be with fetching the item (or as a separate call somehow) rather than in the serialization code. Maybe it should be moved in a later PR?

Reply:
Yeah, absolutely agreed.  We can continue to refine this as we go.